resty/ipmatcher.lua

```diff
@@ -170,40 +170,72 @@ local function gen_ipv6_idxs(inets_ipv6, mask)
 end
 
 
-function _M.new(ips)
+local function iter_ips_with_value(ips)
+    return pairs(ips)
+end
+
+local function iter_ips(ips)
+    return ipairs(ips)
+end
+
+local function new(ips, with_value)
     if not ips or type(ips) ~= "table" then
         error("missing valid ip argument", 2)
     end
 
     local parsed_ipv4s = {}
     local parsed_ipv4s_mask = {}
+    local ipv4_match_all_value
+
     local parsed_ipv6s = {}
     local parsed_ipv6s_mask = {}
+    local ipv6_values = {}
+    local ipv6s_values_idx = 1
+    local ipv6_match_all_value
+
+    local iter = with_value and iter_ips_with_value or iter_ips
```

Comment:
This makes me a little confused, why not just use `pairs` or `ipairs ` directly?